fix(import/ova): speedup the import of gziped vmdk disks nested in .ova

nraynaud opened the pull request:
This the followup to #5085

Avoid unzipping the entire file from the beginning before each read.
The test case when from 10min down to 26 seconds.

When reading a block from the gzipped file, we keep the current state in memory, if the next read happens at an offset greater than the previous read, we just carry one decompressing the file until the desired position.

The previous code would decompress from the start of the file for every read operation.

### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [ ] enhancement/bug fix entry added
  - [x] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [x] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

nraynaud committed:
VM import: speedup the import of gziped vmdk disks nested in .ova

Avoid unzipping the entire file from the beginning before each read.

nraynaud commented:
Is it worthy of a changelog entry?

olivierlambert commented:
This kind of improvement is really great! So first, congrats 👍 
And yes, worth a changelog entry with some explanation in our blog post release on how we improved a LOT gziped vmdk in OVAs!

julien-f reviewed (commented):
Waiting for changelog entry.

nraynaud committed:
simplify gzip handler

Do not move the zstream after we pulled our last chunk. This allows the next read() operation to re-use the stream even if it starts slightly before the end of the previous read().

nraynaud committed:
Merge branch 'master' into nr-speedup-gzip-ova

nraynaud committed:
add changelog

nraynaud committed:
code style

julien-f committed:
Merge branch 'master' into nr-speedup-gzip-ova

julien-f committed:
fix(CHANGELOG.unreleased): vmdk → VMDK

nraynaud committed:
Merge branch 'master' into nr-speedup-gzip-ova
